packages/preview/src/browser/preview-widget.ts

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import {
+    inject,
+    injectable
+} from "inversify";
+import {
+    Resource,
+    DisposableCollection
+} from '@theia/core';
+import {
+    BaseWidget,
+    Message,
+    StatefulWidget
+} from '@theia/core/lib/browser';
+import URI from '@theia/core/lib/common/uri';
+import {
+    ResourceProvider,
+    Disposable
+} from '@theia/core/lib/common';
+import {
+    Workspace,
+    TextDocument,
+    DidChangeTextDocumentParams
+} from "@theia/languages/lib/common";
+import {
+    PreviewHandler,
+    PreviewHandlerProvider
+} from './preview-handler';
+
+export const PREVIEW_WIDGET_CLASS = 'theia-preview-widget';
+
+export const PREVIEW_WIDGET_FACTORY_ID = 'preview-widget';
+
+@injectable()
+export class PreviewWidget extends BaseWidget implements StatefulWidget {
+
+    protected resource: Resource | undefined;
+    protected previewHandler: PreviewHandler | undefined;
+    protected readonly resourceDisposibles = new DisposableCollection();
+
+    @inject(ResourceProvider)
+    protected readonly resourceProvider: ResourceProvider;
+
+    @inject(Workspace)
+    protected readonly workspace: Workspace;
+
+    @inject(PreviewHandlerProvider)
+    protected readonly previewHandlerProvider: PreviewHandlerProvider;
+
+    constructor(
+    ) {
+        super();
+        this.id = 'preview';
+        this.title.closable = true;
+        this.addClass(PREVIEW_WIDGET_CLASS);
+        this.node.tabIndex = 0;
+        this.startScrollSync();
+        this.update();
+    }
+
+    protected scrollSyncTimer: number | undefined = undefined;
+    protected startScrollSync(): void {
+        this.node.addEventListener('scroll', event => {
+            if (this.scrollSyncTimer) {
+                window.clearTimeout(this.scrollSyncTimer);
+            }
+            this.scrollSyncTimer = window.setTimeout(() => {
+                const scrollTop = this.node.scrollTop;
+                this.didScroll(scrollTop);
+            }, 200);
+        });
+    }
+
+    onActivateRequest(msg: Message): void {
+        super.onActivateRequest(msg);
+        this.update();
```

Comment:
closeable widgets should focus something on the activation

Reply:
not sure about this. the scenario is: the user is opening the preview to the side of current editor, and continues to modify source in editor. blurring editor would mean, that the user needs to select the editor again.

Reply:
ok, makes sense

Reply:
@spoenemann does that cause any trouble, if the widget doesn't get tracked?

Reply:
Thanks @spoenemann! I've fixed this.